examples/carousel/carousel-1/js/carousel.js

```diff
@@ -0,0 +1,209 @@
+    }
+    this.lastItem = item;
+
+    var imageLinks = items[i].querySelectorAll('.carousel-image a');
+
+    if (imageLinks && imageLinks[0]) {
+      imageLinks[0].addEventListener('focus', this.handleImageLinkFocus.bind(this));
+      imageLinks[0].addEventListener('blur', this.handleImageLinkBlur.bind(this));
+    }
+
+  }
+
+  // Next Slide and Previous Slide Buttons
+
+  var elems = document.querySelectorAll('.carousel a.carousel-control');
+
+  for (var i = 0; i < elems.length; i++) {
+    if (elems[i].tagName.toLowerCase() == 'a') {
+      var button = new CarouselButton(elems[i], this);
+
+      button.init();
+    }
+  }
+
+  this.currentItem = this.firstItem;
+
+  this.pauseButton = this.domNode.parentNode.parentNode.querySelector('button.pause');
+  if (this.pauseButton) {
+    var button = new PauseButton(this.pauseButton, this);
+    button.init();
+    this.pauseButton.innerHTML = this.stopLabel;
+  }
+
+  this.domNode.addEventListener('mouseover', this.handleMouseOver.bind(this));
+  this.domNode.addEventListener('mouseout', this.handleMouseOut.bind(this));
+
+  // Start rotation
+  setTimeout(this.rotateSlides.bind(this), this.timeInterval);
+};
+
+Carousel.prototype.setSelected = function (newItem, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  for (var i = 0; i < this.items.length; i++) {
+    this.items[i].hide();
+  }
+
+  this.currentItem = newItem;
+  this.currentItem.show();
+
+  if (moveFocus) {
+    this.currentItem.domNode.focus();
+  }
+};
+
+Carousel.prototype.setSelectedToPreviousItem = function (currentItem, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  var index;
+
+  if (typeof currentItem !== 'object') {
+    currentItem = this.currentItem;
+  }
+
+  if (currentItem === this.firstItem) {
+    this.setSelected(this.lastItem, moveFocus);
+  }
+  else {
+    index = this.items.indexOf(currentItem);
+    this.setSelected(this.items[index - 1], moveFocus);
+  }
+};
+
+Carousel.prototype.setSelectedToNextItem = function (currentItem, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  var index;
+
+  if (typeof currentItem !== 'object') {
+    currentItem = this.currentItem;
+  }
+
+  if (currentItem === this.lastItem) {
+    this.setSelected(this.firstItem, moveFocus);
+  }
+  else {
+    index = this.items.indexOf(currentItem);
+    this.setSelected(this.items[index + 1], moveFocus);
+  }
+};
+
+Carousel.prototype.rotateSlides = function () {
+  if (this.rotate) {
+    this.setSelectedToNextItem();
+  }
+  setTimeout(this.rotateSlides.bind(this), this.timeInterval);
+};
+
+Carousel.prototype.startRotation = function () {
+  if (!this.hasHover && !this.hasFocus && !this.isStopped) {
+    this.rotate = true;
+    this.liveRegionNode.setAttribute('aria-live', 'off');
+    this.pauseButton.innerHTML = this.stopLabel;
+  }
+  this.pauseButton.disabled = this.hasHover | this.hasFocus;
```

Comment:
@jongund, I think we should use aria-disabled and keep the button focusable when disabled so SR users can hear the state of the carousel when tabbing.